### ISSUE TITLE
Not yet incompatible with IPFS 0.4.0

### ISSUE THREAD

**Issue by Kubuxu** [diasdavid/ipscend#27]
It published this: https://ipfs.io/ipfs/QmWu8wbhogrQHPiLgv1Ed7iM5UaH4ZQ4zveZg89Yce554n instead of the site.

**Comment by jbenet** [comment]
@diasdavid @whyrusleeping looks like our favorite thing in the world. Multipart! \o/

**Comment by Kubuxu** [comment]
To compensate I made this: https://ipfs.io/ipns/bin.ipfs.ovh/#QmVM9iUVfZbyC2bNyzjij6oVofPWH1PAjepDTFpp8DsCuJ
It is dumb but it works.

**Comment by daviddias** [comment]
@jbenet [ahahahahah](http://koruza.net/assets/images/features/img1.jpg)

@Kubuxu my intention is to revisit ipscend when we release 0.4, currently 0.4 http API differs a bit and we do have indeed a js-ipfs-api branch that works well with it (here https://github.com/ipfs/js-ipfs-api/pull/74)

**Comment by Kubuxu** [comment]
My proposition for usage of files API:
1. During initial configuration ask user to which directory in files API he wants to publish.
2. Add content to timestamped subdirectory and copy (copy is zero cost) and replace directory `current` under this directory.

I propose ISO 8601 compatible timestamp format: `$(date -u +%FT%TZ) == 2016-02-22T15:59:01Z`
Created directory structure is:

```
$ ipfs files ls /public/ovh-kubuxu
2016-02-11T09:35:41Z
2016-02-17T14:43:51Z
current
```

Where current is alias for the most recently released version.

Printing hash of current release, updated target directory and user chosen directory would be nice (in my script I have config for that and I usually print `/public/` as I publish it then to IPNS).

This structure is very IPNS/DNS friendly as it allows to hold multiple projects under one IPNS name. Example:

```
www.kubuxu.ovh    TXT    dnslink=/ipns/QmAAA...AAA/ovh-kubuxu/current
bin.ipfs.ovh      TXT    dnslink=/ipns/QmAAA...AAA/ovh-ipfs-bin/current
```

My script in bash for comparison: https://gist.github.com/Kubuxu/d0c133cbc0ea4693db67

**Comment by olizilla** [comment]
@diasdavid is it this issue still the case, or did this get fixed in https://github.com/diasdavid/ipscend/commit/0889197cd9dfd0473cdeef2d6a077d374b4a6721

**Comment by daviddias** [comment]
Oh, I just really need to update this module. Let me do that today 🚀

**Comment by daviddias** [comment]
Just updated the module (and all of its deps) and released 0.4.2. Should be golden now, let me know if you ran into any trouble!
